### PR TITLE
Added rate limiting to api calls

### LINKED ISSUE #249: Handle Rate Limiting of Slack API
It seems that the issues we are having are caused by rate limiting in the Slack Web API: https://api.slack.com/docs/rate-limits

The reason we only see this on our actual slack and not the test slack is because there are so many members that it paginates the results when looping through the channel members, and we are not rate limiting our pagination attempts.

We apparently need to listen for a HTTP 429 Too Many Requests response and that will tell us how long to wait (in seconds!) until we should try again.

### PR DESCRIPTION
**Pull request by mitchmcdee**
As per title, adds rate limit handling to api calls.
Comments and feedback appreciated :^)

Fixes #249.

### CODE REVIEW

File: uqcsbot/api.py [hunk 1]
```diff
@@ -67,17 +68,30 @@ def __call__(self, **kwargs) -> dict:
...
-            loop = asyncio.get_event_loop()
-            return loop.run_in_executor(None, fn)
+        retry_count = 0
+        while retry_count < 5:
+            if self._async:
+                loop = asyncio.get_event_loop()
+                result = loop.run_in_executor(None, fn)
+            else:
+                result = fn()
+            if not result['ok'] and result['error'] == 'ratelimited':
```
Comment (TRManderson) [inline uqcsbot/api.py]: Can't just use result like this - needs to be awaited if `self._async`

Comment (mitchmcdee) [reply uqcsbot/api.py]: like, `result = await loop.run_in_executor(None, fn)`?

Comment (mitchmcdee) [reply uqcsbot/api.py]: like, `result = await loop.run_in_executor(None, fn)`?

**Comment by mitchmcdee** [comment]
re: `Try doing an inner async function that does the looping and passing that in to the loop.run_in_executor call`

for what reason do I need to do that? Isn't that going to look super duplicate-y if I need to do the loops for non-async too? or what did you have in mind here?

**Comment by mitchmcdee** [comment]
@TRManderson PTAL <3

**Comment by TRManderson** [comment]
Added a PR that builds on this. Yes it looks hella duplicate-y, but unfortunately that's what you get in Python.

**Pull request merged**